front-end/src/components/Post/Poll/CouncilSignals.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { DeriveElectionsInfo } from '@polkadot/api-derive/types';
+import { ApiPromiseContext } from '@substrate/context';
+import styled from '@xstyled/styled-components';
+import React, { useContext, useEffect, useState } from 'react';
+import { Grid, Icon } from 'semantic-ui-react';
+
+import { PostVotesQuery } from '../../../generated/graphql';
+import { OffchainVote, Vote } from '../../../types';
+import Address from '../../../ui-components/Address';
+import Card from '../../../ui-components/Card';
+import CouncilSignalBar from '../../../ui-components/CouncilSignalBar';
+import getDefaultAddressField from '../../../util/getDefaultAddressField';
+import getEncodedAddress from '../../../util/getEncodedAddress';
+
+interface Props {
+	className?: string
+	data?: PostVotesQuery | undefined
+}
+
+const CouncilSignals = ({ className, data }: Props) => {
+	const [ayes, setAyes] = useState(0);
+	const [nays, setNays] = useState(0);
+	const [memberSet, setMemberSet] = useState<Set<string>>(new Set<string>());
+	const [councilVotes, setCouncilVotes] = useState<OffchainVote[]>([]);
+	const { api, isApiReady } = useContext(ApiPromiseContext);
+
+	useEffect(() => {
+		if (!isApiReady){
+			return;
+		}
+
+		let unsubscribe: () => void;
+
+		api.derive.elections.info((info: DeriveElectionsInfo) => {
+			const memberSet = new Set<string>();
+
+			info?.members.map(([accountId]) => {
+				const address = getEncodedAddress(accountId.toString()) || '';
+				memberSet.add(address);
+			});
+
+			setMemberSet(memberSet);
+		})
+			.then(unsub => { unsubscribe = unsub; })
+			.catch(e => console.error(e));
+
+		return () => unsubscribe && unsubscribe();
+	}, [api, isApiReady]);
+
+	useEffect(() => {
+		let ayes = 0;
+		let nays = 0;
+		const defaultAddressField = getDefaultAddressField();
+		const councilVotes: OffchainVote[]  = [];
+
+		data?.post_votes?.forEach(({ vote, voter }) => {
+			const defaultAddress = voter?.[defaultAddressField] || '';
+
+			if (memberSet.has(defaultAddress)) {
+				const address = getEncodedAddress(defaultAddress);
+				if (address) {
+					councilVotes.push({
+						address,
+						vote
+					});
+				}
+
+				if (vote === Vote.AYE) {
+					ayes++;
+				}
+
+				if (vote === Vote.NAY) {
+					nays++;
+				}
+			}
+		});
+
+		setAyes(ayes);
+		setNays(nays);
+		setCouncilVotes(councilVotes);
+	}, [data, memberSet]);
+
+	return (
+		<Card className={className}>
+			<h3>Council Signals</h3>
+			<CouncilSignalBar
+				ayeSignals={ayes}
+				naySignals={nays}
```

Comment:
Should probably make `councilSize` non-optional in `CouncilSignalBar` and display a `Loader` till it got the number from the api
```suggestion
				councilSize={memberSet.size}
				naySignals={nays}
```

Reply:
Obviously a temporary fix if we're checking number of council members at a current block later on